zulipterminal/ui_tools/boxes.py

```diff
@@ -107,27 +107,30 @@ def autocomplete_mentions(self, text: str, state: int,
                               prefix_string: str) -> Optional[str]:
         # Handles user mentions (@ mentions and silent mentions)
         # and group mentions.
+        at_index = max(text.find(prefix_string), 0)
         group_typeahead = ['@*{}*'.format(group_name)
                            for group_name in self.model.user_group_names
-                           if match_groups(group_name, text[1:])]
+                           if match_groups(group_name, text[at_index+1:])]
 
         users_list = self.view.users
         user_typeahead = [prefix_string+'**{}**'.format(user['full_name'])
                           for user in users_list
-                          if match_user(user, text[len(prefix_string):])]
+                          if match_user(user,
+                                        text[len(prefix_string)+at_index:])]
         combined_typeahead = group_typeahead + user_typeahead
         try:
-            return combined_typeahead[state]
+            return text[:at_index] + combined_typeahead[state]
         except (IndexError, TypeError):
             return None
 
     def autocomplete_streams(self, text: str, state: int) -> Optional[str]:
+        hashtag_index = max(text.find('#'), 0)
```

Comment:
Changing the `autocomplete_streams`' signature would require its test to be amended as well which is not related to this PR (if that makes sense). I think this can be a follow-up though. 
@neiljp What are your thoughts on this?

Reply:
Hmm, since it's a minor change, maybe you could add it in a separate commit before this one? 
This will not only separate unrelated changes but will also ensure that the current commit is of better quality. Just throwing out suggestions, feel free to do a follow-up if you like :)

Reply:
I was suggesting a follow-up for the sake of easiness in terms of reviewing.

Reply:
Well, PRs are generally reviewed commit-wise, so I don't think that should be a problem either :)

Reply:
Alright, I will push the changes.